src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -396,3 +401,42 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
             cpu_offload=True,
         )
         cls._registered_strategies.append("fsdp_cpu_offload")
+
+    def load_optimizer_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
+        optimizer_states = checkpoint["optimizer_states"]
+
+        assert self.model is not None
+        assert isinstance(self.model, FullyShardedDataParallel)
+
+        # rank0_only should be false because we need to load the optimizer state on all ranks
+        with _get_full_state_dict_context(self.model, rank0_only=False):
+            for optimizer, opt_state in zip(self.optimizers, optimizer_states):
```

Comment:
Should we check that the lengths match? In case one reloads with a different optimizer configuration

Reply:
Should it raise an error when users attempt to load different optimizer configs? Or should we add a warning here?